spec/unit/pops/loaders/loaders_spec.rb

```diff
@@ -123,6 +156,68 @@
     expect(loaders.private_environment_loader().to_s).to eql("(DependencyLoader 'environment private' [])")
   end
 
+  it 'creates a hierarchy of loaders' do
+    expect_loader_hierarchy(
+      Puppet::Pops::Loaders.new(empty_test_env),
+      [
+        'environment private',
+        'environment',
+        empty_test_env.name,
+        'puppet_system',
+        nil
+      ],
+      [
+        Puppet::Pops::Loader::DependencyLoader,
+        Puppet::Pops::Loader::SimpleEnvironmentLoader,
+        Puppet::Pops::Loader::Runtime3TypeLoader,
+        Puppet::Pops::Loader::ModuleLoaders::LibRootedFileBased,
+        Puppet::Pops::Loader::StaticLoader
```

Comment:
It's a little confusing the way this is laid out, I'm thinking it would be clearer to list the `name, class` pairs in order from the top (StaticLoader) to the leaf, eg

```
expect_loader_hierarchy(
  Puppet::Pops::Loaders.new(empty_test_env),
  [
    [nil,              Puppet::Pops::Loader::StaticLoader],
    ['puppet_system',  Puppet::Pops::Loader::ModuleLoaders::LibRootedFileBased],
    ...
  ]
```

Reply:
Updated